log4j2-propagation/src/test/java/nl/talsmasoftware/context/log4j2/threadcontext/Log4J2ThreadContextSnapshotTest.java

```diff
@@ -15,93 +15,97 @@
-        assertTrue(ThreadContext.isEmpty());
-
-        Log4j2ThreadContextData data = Log4j2ThreadContextData.fromCurrentThreadContext();
-        assertTrue(data.getContextMap().isEmpty());
-        assertTrue(data.getContextStack().isEmpty());
-
-        // Applying empty data should have no effect
-        data.applyToCurrentThread();
-        assertEquals(0, ThreadContext.getDepth());
-        assertTrue(ThreadContext.isEmpty());
+    void testCaptureFromCurrentThread_empty() {
+        assertThat(ThreadContext.getDepth(), is(0));
+        assertThat(ThreadContext.isEmpty(), is(true));
+
+        Log4j2ThreadContextSnapshot snapshot = Log4j2ThreadContextSnapshot.captureFromCurrentThread();
+        assertThat(snapshot.getContextMap(), is(anEmptyMap()));
+        assertThat(snapshot.getContextStack(), is(empty()));
+
+        // Applying empty snapshot should have no effect
+        snapshot.applyToCurrentThread();
+        assertThat(ThreadContext.getDepth(), is(0));
+        assertThat(ThreadContext.isEmpty(), is(true));
     }
 
     @Test
     void testFromCurrentThreadContext_empty_apply() {
-        assertEquals(0, ThreadContext.getDepth());
-        assertTrue(ThreadContext.isEmpty());
+        assertThat(ThreadContext.getDepth(), is(0));
+        assertThat(ThreadContext.isEmpty(), is(true));
 
-        Log4j2ThreadContextData data = Log4j2ThreadContextData.fromCurrentThreadContext();
+        Log4j2ThreadContextSnapshot snapshot = Log4j2ThreadContextSnapshot.captureFromCurrentThread();
 
         ThreadContext.put("map1", "value1");
         ThreadContext.put("map2", "value2");
         ThreadContext.push("stack1");
         ThreadContext.push("stack2");
 
         // Modification of ThreadContext should not have affected snapshot
-        assertTrue(data.getContextMap().isEmpty());
-        assertTrue(data.getContextStack().isEmpty());
+        assertThat(snapshot.getContextMap(), anEmptyMap());
+        assertThat(snapshot.getContextStack(), empty());
 
         // Applying empty context on top of existing one should have no effect
-        data.applyToCurrentThread();
+        snapshot.applyToCurrentThread();
 
         Map<String, String> expectedMap = new HashMap<String, String>();
         expectedMap.put("map1", "value1");
         expectedMap.put("map2", "value2");
-        assertEquals(expectedMap, ThreadContext.getContext());
+        assertThat(ThreadContext.getContext(), equalTo(expectedMap));
 
         List<String> expectedStack = Arrays.asList("stack1", "stack2");
-        assertEquals(expectedStack, ThreadContext.getImmutableStack().asList());
+        assertThat(ThreadContext.getImmutableStack().asList(), equalTo(expectedStack));
     }
 
     /**
-     * Verify that {@link Log4j2ThreadContextData} is a snapshot and not affected
+     * Verify that {@link Log4j2ThreadContextSnapshot} is a snapshot and not affected
      * by subsequent modification of {@link ThreadContext}.
      */
     @Test
```

Comment:
```diff
-    void testFromCurrentThreadContext_modification() {
+    void testCaptureFromCurrentThread_modification() {
```
(couldn't use GitHub suggestion comment here because it is not part of changed code)